library/classes/Document.class.php

```diff
@@ -695,6 +875,87 @@ function createDocument(
         return '';
     }
 
+    /**
+     * Retrieves the document data that has been saved to the filesystem or couch db.  If the $force_no_decrypt flag is
+     * set to true, it will return the encrypted version of the data for the document.
+     * @param bool $force_no_decrypt True if the document should have its data returned encrypted, false otherwise
+     * @throws BadMethodCallException Thrown if the method is called when the document has been marked as deleted or expired
+     * @return false|string Returns false if the data failed to decrypt, or a string if the data decrypts or is unencrypted.
+     */
+    function get_data($force_no_decrypt = false)
+    {
+        $storagemethod = $this->get_storagemethod();
+
+        if ($this->has_expired()) {
+            throw new BadMethodCallException("Should not attempt to retrieve data from expired documents");
+        }
+        if ($this->is_deleted()) {
+            throw new BadMethodCallException("Should not attempt to retrieve data from deleted documents");
+        }
+
+        $base64Decode = false;
+
+        if ($storagemethod === self::STORAGE_METHOD_COUCHDB) {
+            $base64Decode = true;
+            // Taken from ccr/display.php
+            $couch_docid = $this->get_couch_docid();
+            $couch_revid = $this->get_couch_revid();
+            $couch = new CouchDB();
+            $resp = $couch->retrieve_doc($couch_docid);
+            $data = $resp->data;
+        } else {
+            $data = $this->get_content_from_filesystem();
+        }
+
+        if (!empty($data)) {
+            if ($this->is_encrypted() && !$force_no_decrypt) {
+                $data = $this->decrypt_content($data);
+            }
```

Comment:
will need to change above 3 lines to something like this to get couchdb to work correctly:
```
            if ($this->is_encrypted()) {
                    // couchdb is not base64 encoded when it is encrypted, so need to force below flag to false
                    $base64Decode = false;
                    if (!$force_no_decrypt) {
                        $data = $this->decrypt_content($data);
                    }
            }
```

Reply:
This should now be fixed.